src/dune_named_pipe_stubs/named_pipe.c

```diff
@@ -0,0 +1,143 @@
+#include <caml/bigarray.h>
+#include <caml/callback.h>
+#include <caml/fail.h>
+#include <caml/memory.h>
+#include <caml/mlvalues.h>
+#include <caml/threads.h>
+#include <caml/unixsupport.h>
+#include <stdint.h>
+
+#ifdef WIN32
+#include <stdio.h>
+#include <tchar.h>
+#include <winbase.h>
+#include <winsock2.h>
+#include <wtypes.h>
+
+#endif
+
+static void named_pipe_not_available() {
+  caml_failwith("Named pipes not available");
+}
+
+/* string -> t */
+CAMLprim value stub_named_pipe_create(value path) {
+  CAMLparam1(path);
+  CAMLlocal1(result);
+#ifdef WIN32
+  char *c_path = strdup(String_val(path));
+  HANDLE h = INVALID_HANDLE_VALUE;
+  DWORD nOutBufferSize = 4096;
+  DWORD nInBufferSize = 4096;
+  DWORD nDefaultTimeOut = 0;
+  LPSECURITY_ATTRIBUTES lpSecurityAttributes = NULL;
+  caml_release_runtime_system();
```

Comment:
I think the release/acquire of the runtime system will vastly exceed the cost of `CreateNamedPipeW`!